modules/azure/compute.go

```diff
@@ -8,97 +8,451 @@ import (
+	if len(vmNICs) == 0 {
+		// No NIC present
+		return nil, nil
+	}
 
-	// Create an authorizer
-	authorizer, err := NewAuthorizer()
+	// Get the Names of the attached NICs
+	nics := make([]string, len(vmNICs))
+
+	for i, nic := range vmNICs {
+		nics[i] = GetNameFromResourceID(*nic.ID)
+	}
+	return nics, nil
+}
+
+// GetVirtualMachineNicCount gets the Network Interface count of the specified Azure Virtual Machine.
+// This function would fail the test if there is an error.
+func GetVirtualMachineNicCount(t testing.TestingT, vmName string, resGroupName string, subscriptionID string) int {
+	nicCount, err := GetVirtualMachineNicCountE(vmName, resGroupName, subscriptionID)
+	require.NoError(t, err)
+
+	return nicCount
+}
+
+// GetVirtualMachineNicCountE gets the Network Interface count of the specified Azure Virtual Machine.
+func GetVirtualMachineNicCountE(vmName string, resGroupName string, subscriptionID string) (int, error) {
+	// Get VM Object
+	vm, err := GetVirtualMachineE(vmName, resGroupName, subscriptionID)
+	if err != nil {
+		return -1, err
+	}
+
+	return len(*vm.NetworkProfile.NetworkInterfaces), nil
+}
+
+// GetVirtualMachineManagedDisks gets the list of Managed Disk names of the specified Azure Virtual Machine.
+// This function would fail the test if there is an error.
+func GetVirtualMachineManagedDisks(t testing.TestingT, vmName string, resGroupName string, subscriptionID string) []string {
+	diskNames, err := GetVirtualMachineManagedDisksE(vmName, resGroupName, subscriptionID)
+	require.NoError(t, err)
+
+	return diskNames
+}
+
+// GetVirtualMachineManagedDisksE gets the list of Managed Disk names of the specified Azure Virtual Machine.
+func GetVirtualMachineManagedDisksE(vmName string, resGroupName string, subscriptionID string) ([]string, error) {
+
+	// Get VM Object
+	vm, err := GetVirtualMachineE(vmName, resGroupName, subscriptionID)
 	if err != nil {
 		return nil, err
 	}
 
-	// Attach authorizer to the client
-	vmClient.Authorizer = *authorizer
+	// Get VM attached Disks
+	vmDisks := *vm.StorageProfile.DataDisks
+
+	// No Attached Disks present
+	if len(vmDisks) == 0 {
+		return nil, nil
+	}
+
+	// Get the Names of the attached Managed Disks
+	diskNames := make([]string, len(vmDisks))
+	for _, v := range vmDisks {
+		diskNames = append(diskNames, *v.Name)
+	}
 
-	return &vmClient, nil
+	return diskNames, nil
 }
 
-// GetSizeOfVirtualMachine gets the size type of the given Azure Virtual Machine.
+// GetVirtualMachineManagedDiskCount gets the Managed Disk count of the specified Azure Virtual Machine.
 // This function would fail the test if there is an error.
-func GetSizeOfVirtualMachine(t testing.TestingT, vmName string, resGroupName string, subscriptionID string) compute.VirtualMachineSizeTypes {
-	size, err := GetSizeOfVirtualMachineE(t, vmName, resGroupName, subscriptionID)
+func GetVirtualMachineManagedDiskCount(t testing.TestingT, vmName string, resGroupName string, subscriptionID string) int {
```

Comment:
NIT: Same question as above re: reusing the other function.

Reply:
Reused, good point!